hbase-server/src/main/java/org/apache/hadoop/hbase/replication/ChainWALEntryFilter.java

```diff
@@ -56,6 +59,13 @@ public ChainWALEntryFilter(List<WALEntryFilter> filters) {
     initCellFilters();
   }
 
+  public ChainWALEntryFilter(List<WALEntryFilter> filters, String operatorName) {
+    this(filters);
+    if (!StringUtils.isEmpty(operatorName)) {
+      this.operator = Operator.valueOf(operatorName);
```

Comment:
I was trying to figure out the first place we read the String "operatorName" and make sure it fails gracefully.

I know you have the client-side checking in Ruby code, and I suggested we have Java data validation. We should check it here as future-proofing. I think this happens early enough in the replication setup that the client would see a RemoteException flowing back to them? (not that their `add_peer` call would succeed and just not replicate any data because it failed).

Reply:
You mean, beyond the checks to avoid an NPE, explicitly extra check for the valid strings and throw IllegalArgumentException, rather than letting the enum error?

Reply:
Yeah, something that we would at least throw back a well-formed exception (and not something that might be very terse/short)